### PR TITLE
[PLGN-608] Cisco Firepower Management Center - `Check if Address in Group`: Extended search for manually added literals | Added new output field literal_objects

### PR DESCRIPTION
**Pull request by igorski-r7**
### Description

Describe the proposed changes:

  - `Check if Address in Group`: Extended search for manually added literals
  - Added new output field literal_objects
  - Refactored unittests
  - Updated the SDK to latest version using Python `3.9.18` slim
  - Updated dependencies

Current response for `Check if Address in Group` action contains new additional field `literal_objects`. With `enable_search = false`:

```
{
  "body": {
    "log": "",
    "status": "ok",
    "meta": {},
    "output": {
      "found": true,
      "address_objects": [],
      "literal_objects": [
        {
          "type": "Network",
          "value": "192.168.0.0/16"
        }
      ]
    }
  },
  "version": "v1",
  "type": "action_event"
}
```

Setting up `enable_search = true` allows to lookup for address_objects. If any is found it additionally returns the address_objects.

## PR Requirements

Developers, verify you have completed the following items by checking them off:

#### Unit Tests

Review our documentation on [generating](https://docs.rapid7.com/insightconnect/unit-test-generation) and [writing](https://docs.rapid7.com/insightconnect/unit-test-primer) plugin unit tests

- [X] Unit tests written for any new or updated code

#### In-Product Tests

If you are an InsightConnect customer or have access to an InsightConnect instance, the following in-product tests should be done:

- [X] Screenshot of job output with the plugin changes
- [X] Screenshot of the changed connection, actions, or triggers input within the InsightConnect workflow builder

### Style

Review the [style guide](https://docs.rapid7.com/insightconnect/style-guide/)

- [X] For dependencies, pin [OS package](https://docs.rapid7.com/insightconnect/style-guide/#dockerfile) and [Python package](https://docs.rapid7.com/insightconnect/style-guide/#requirements.txt) versions
- [X] For security, set least privileged account with ``USER nobody`` in the ``Dockerfile`` when possible
- [X] For size, use the [slim SDK images](https://docs.rapid7.com/insightconnect/sdk-guide/#sdk-guide) when possible: ``rapid7/insightconnect-python-3-38-slim-plugin:{sdk-version-num}`` and ``rapid7/insightconnect-python-3-38-plugin:{sdk-version-num}``
- [X] For error handling, use of [PluginException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations/#plugin-exceptions) and [ConnectionTestException](https://docs.rapid7.com/insightconnect/error-handling-in-integrations#connection-exceptions)
- [X] For logging, use [self.logger](https://docs.rapid7.com/insightconnect/sdk-guide/#logging)
- [X] For docs, use [changelog style](https://docs.rapid7.com/insightconnect/style-guide/#changelog)
- [X] For docs, validate markdown with ``insight-plugin validate`` which calls ``icon_validate`` to lint ``help.md``

### Functional Checklist
- [X] Work fully completed
- [X] Functional
  - [X] Any new actions/triggers include JSON [test files](https://docs.rapid7.com/insightconnect/style-guide/#tests) in the `tests/` directory created with `insight-plugin samples`
  - [X] Tests should all pass unless it's a negative test. Negative tests have a naming convention of `tests/$action_bad.json`
  - [X] Unsuccessful tests should fail by raising an exception causing the plugin to die and an object should be returned on successful test
  - [X] Add functioning test results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -T tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run -T all --debug --jq` (use PR format at end)
  - [X] Add functioning run results to PR, sanitize any output if necessary
    * Single action/trigger `insight-plugin run -R tests/example.json --debug --jq`
    * All actions/triggers shortcut `insight-plugin run --debug --jq` (use PR format at end)

### Assessment

You must validate your work to reviewers:

1. Run `insight-plugin validate` and make sure everything passes
2. Run the assessment tool: `insight-plugin run -A`. For single action validation: `insight-plugin run tests/{file}.json -A`
3. Copy (`insight-plugin ... | pbcopy`) and paste the output in **a new post** on this PR
4. Add required screenshots from the In-Product Tests section

### CODE REVIEW

File: plugins/cisco_firepower_management_center/requirements.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
...
 # See: https://pip.pypa.io/en/stable/user_guide/#requirements-files
-fmcapi==20211214.0
-validators==0.18.2
-pyOpenSSL==23.0.0
-parameterized==0.8.1
\ No newline at end of file
+fmcapi==20231211.0
+pyOpenSSL==23.3.0
+validators==0.22.0
+parameterized==0.9.0
```
Comment (ablakley-r7) [inline plugins/cisco_firepower_management_center/requirements.txt:7]: Just checking if we need to pin cryptography here?

**Review by ablakley-r7 — APPROVED** [review]

File: plugins/cisco_firepower_management_center/requirements.txt (hunk 1 above)
Comment (joneill-r7) [reply plugins/cisco_firepower_management_center/requirements.txt:7]: Rest of the changes look good, happy to approve once this comment is resolved. Looking at the dependancies for pyopenssl the snyk vulns should be addressed now? But could be worth pinning as Adam mentioned?

Comment (dlaverty-r7) [reply plugins/cisco_firepower_management_center/requirements.txt:7]: Agreed, think cryptography will need pinned to 41.0.2 or the vulnerability will remain.

Comment (igorski-r7) [reply plugins/cisco_firepower_management_center/requirements.txt:7]: The latest `pyOpenSSL==23.3.0` installs latest version of `cryptography==41.0.7`. This is why I haven't pinned it there 😄

**Review by dlaverty-r7 — APPROVED** [review]

**Pull request merged**